tools/atlas-api-automation/differ.py

```diff
@@ -0,0 +1,107 @@
+
+    # must provide path to local cloud-docs dir
+    if len(args) == 0:
+        print('Please provide a path to the cloud-docs directory.')
+        exit()
+
+    ATLAS_API_REFERENCE = args[0] + '/source/reference/api/'
+
+    # cloud-docs dir provided must contain api files
+    if not isdir(ATLAS_API_REFERENCE):
+        err = '''\nThe path: \n{apiref}\ndoes not contain a list of reference api files for atlas.
+        '''.format(apiref=ATLAS_API_REFERENCE)
+        print(err)
+        exit()
+
+    # get files in reference api directory for atlas
+    api_files = [f for f in listdir(ATLAS_API_REFERENCE) if isfile(join(ATLAS_API_REFERENCE, f))]
+
+    # will do a diff on these lists to see what new endpoints have not been documented yet
+    atlas_endpoints_in_docs = {}
+    atlas_endpoints_diff_from_docs = []
+
+    # store endpoints we currently have for atlas docs in dictionary
+    for api_file in api_files:
+        with open(ATLAS_API_REFERENCE + api_file) as file_contents:
+            regex = re.search(ROUTE_REGEX, file_contents.read())
+            # if API route was found in text, replace base url and params with wildcard chars
+            if regex:
+                route_string = re.sub(r'(\/?api\/atlas\/v1\.0)', '', regex.group())
+                route_string = re.sub(r'{.+?}', '{*}', route_string)
+                route_string = route_string.replace('{*}:{*}', '{*}')
+                # some routes end with a trailing slash and so correct routes won't match
+                # e.g. 'GET /orgs/{*}/invoices' will not match 'GET /orgs/{*}/invoices/'
+                if route_string[-1] == '/':
+                    route_string = route_string[:-1]
+                # pp.pprint(route_string)
+                atlas_endpoints_in_docs[route_string] = True
+
+    # get json data of latest atlas code
+    with open(ATLAS_JSON_FILE) as atlas_json:
+        data = json.load(atlas_json)
+        for route in data:
+            route_path = route['path']
+            # if this route is part of atlas
+            if route_path.find('atlas') != -1:
+                route_path = route_path.replace('/api/atlas/v1.0', '')
+                route_path = route_path.replace('/api/{app:public|atlas}/v1.0', '')
+                route_path = re.sub('{.+?}', '{*}', route_path)
+                if route_path != '':
+                    route_string = route['method'] + ' ' + route_path
+                    # pp.pprint(route_string)
+                    if route_string not in atlas_endpoints_in_docs:
+                        atlas_endpoints_diff_from_docs.append(route_string)
+
+    # get whitelist of endpoints that we've deemed OK and do not include in output
+    with open(WHITELIST_FILE) as whitelist:
+        SKIP_ENDPOINTS = whitelist.read().splitlines()
+
+    # save the diff list to a file
+    with open(OUTPUT_FILE, 'w') as out_file:
+        written_to_output = 0
+        for endpoint in atlas_endpoints_diff_from_docs:
```

Comment:
Once you have the allow list created before creating `atlas_endpoints_diff_from_docs`, you can apply the allow list while creating it, and get rid of this loop:

`out_file.write('\n'.join(atlas_endpoints_diff_from_docs)`

And `written_to_output` becomes just `len(atlas_endpoints_diff_from_docs)`